src/test/java/org/gridsuite/directory/server/DirectoryTest.java

```diff
@@ -170,6 +170,163 @@ public void testTwoUsersTwoPublicDirectories() {
+
+        UUID unknownUuid = UUID.randomUUID();
+
+        webTestClient.put()
+            .uri("/v1/elements/" + unknownUuid + "?newDirectory=" + rootDir20Uuid)
+            .header("userId", "Doe")
+            .exchange()
+            .expectStatus().isNotFound();
+
+        webTestClient.put()
+            .uri("/v1/elements/" + filterUuid + "?newDirectory=" + unknownUuid)
+            .header("userId", "Doe")
+            .exchange()
+            .expectStatus().isNotFound();
+    }
+
+    @Test
+    public void testMoveElementWithAlreadyExistingNameAndTypeInDestination() {
+        // Insert public root20 directory
+        UUID rootDir20Uuid = insertAndCheckRootDirectory("rootDir20", false, "Doe");
+
+        // Insert a filter in this directory
+        UUID filterUuid = UUID.randomUUID();
+        ElementAttributes filterAttributes = toElementAttributes(filterUuid, "filter", FILTER, null, "Doe");
+        insertAndCheckSubElement(rootDir20Uuid, false, filterAttributes);
+
+        // Insert a subDirectory20 in the root20 directory
+        UUID directory21UUID = UUID.randomUUID();
+        ElementAttributes directory20Attributes = toElementAttributes(directory21UUID, "directory20", DIRECTORY, false, "Doe");
+        insertAndCheckSubElement(rootDir20Uuid, false, directory20Attributes);
+
+        // Insert a filter in the last subdirectory with the same name and type as the 1st one
+        UUID filterwithSameNameAndTypeUuid = UUID.randomUUID();
+        ElementAttributes filterwithSameNameAndTypeAttributes = toElementAttributes(filterwithSameNameAndTypeUuid, "filter", FILTER, null, "Doe");
+        insertAndCheckSubElement(directory21UUID, false, filterwithSameNameAndTypeAttributes);
+
+        webTestClient.put()
+            .uri("/v1/elements/" + filterwithSameNameAndTypeUuid + "?newDirectory=" + rootDir20Uuid)
+            .header("userId", "Doe")
+            .exchange()
+            .expectStatus().isForbidden();
+    }
+
+    @Test
+    public void testMoveElementToNotDirectory() {
+        UUID rootDir20Uuid = insertAndCheckRootDirectory("rootDir20", false, "Doe");
+
+        UUID filter1Uuid = UUID.randomUUID();
+        ElementAttributes filter1Attributes = toElementAttributes(filter1Uuid, "filter1", FILTER, null, "Doe");
+        insertAndCheckSubElement(rootDir20Uuid, false, filter1Attributes);
+
+        UUID filter2Uuid = UUID.randomUUID();
+        ElementAttributes filter2Attributes = toElementAttributes(filter2Uuid, "filter2", FILTER, null, "Doe");
+        insertAndCheckSubElement(rootDir20Uuid, false, filter2Attributes);
+
+        webTestClient.put()
+            .uri("/v1/elements/" + filter1Uuid + "?newDirectory=" + filter2Uuid)
+            .header("userId", "Doe")
+            .exchange()
+            .expectStatus().isForbidden();
+    }
+
+    @Test
+    public void testMoveDirectory() {
+        UUID rootDir10Uuid = insertAndCheckRootDirectory("rootDir10", false, "Doe");
```

Comment:
Add a test to mode root folder. It will be useful !

Reply:
done